doc/nrf/glossary.rst

```diff
@@ -19,42 +56,358 @@ Commit tag
 Contribution
    A change to the codebase sent to a remote repository for inclusion.
 
+Development Kit (DK)
+   A hardware development platform used for application development.
+
+Device Firmware Update (DFU)
+   A mechanism for upgrading the firmware of a device.
+
+Devicetree
+   A data structure for describing hardware and its boot-time configuration, including peripherals on a board, boot-time clock frequencies, interrupt lines, etc.
+
+Devicetree Specification (DTS, DTSpec)
+   A specification that "provides a complete boot program to client program interface definition, combined with minimum system requirements that facilitate the development of a wide variety of systems." Devicetree Specification, Release v0.3.
+
+Device Under Test (DUT)
+   A manufactured product undergoing testing.
+
+Docker
+   A set of software tools using OS-level virtualization to create and run applications and their dependencies in self-contained environments called containers.
+
+Docker container
+   A self-contained environment created by running a Docker container image in the Docker engine.
+
+Docker container image
+   A standard set of binary data that contains an application (or more than one) and all the required dependencies.
+   When run on the Docker engine, it creates a Docker container.
+
+Docker engine
+   The container runtime that runs Docker images.
+
+Domain Name System (DNS)
+   A hierarchical distributed naming system for computers, services, or any resource connected to the Internet or a private network.
+   It associates various information with domain names assigned to each of the participating entities.
+   Most prominently, it translates domain names, which can be easily memorized by humans, to the numerical IP addresses needed for computer services and devices worldwide.
+   The Domain Name System is an essential component of the functionality of most Internet services because it is the Internet's primary directory service.
+
 Downstream
    The repository that is forked off an upstream.
 
+Floating-Point Unit (FPU)
+   A part of a CPU specially designed to perform operations on floating-point numbers.
+
 Fork
    A server-hosted copy of a repository (upstream) that intends to follow the changes made in the original repository as time goes by, while at the same time keeping some other changes unique to it.
 
```

Comment:
Perhaps we should add a glossary entry for [FOTA](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/include/net/fota_download.html?highlight=fota#fota-download)?

Reply:
There is an entry for Over-the-Air (OTA), but I can also add one for FOTA. My only doubt is if it is actually correct, as OTA could be referred to anything happening over-the-air, while FOTA doesn't make sense unless is "FOTA Update". Maybe I can just add this one.

Reply:
Added.

Reply:
Added